### PR TITLE
Script: Enforce type checks if not nil

### LINKED ISSUE #9659: Builtin should sanity-check some properties on node registration
##### Minetest version

```
Minetest 5.3.0-dev-27d611fe5-dirty (Linux)
BUILD_TYPE=Release
RUN_IN_PLACE=0
USE_GETTEXT=0
USE_SOUND=0
USE_CURL=1
USE_FREETYPE=1
USE_LUAJIT=0
STATIC_SHAREDIR="/usr/local/share/minetest"
```

##### OS / Hardware

Operating system:
Docker on centos
CPU:
xeon e5620

GPU model:
OpenGL version:

##### Summary

```
minetest_1  | 2020-04-13 15:23:14: ERROR[Main]: ServerError: AsyncErr: ServerThread::run Lua: Runtime error from mod '*builtin*' in callba
ck luaentity_Activate(): /usr/local/share/minetest/builtin/game/falling.lua:106: attempt to perform arithmetic on field 'visual_scale' (a
table value)
minetest_1  | 2020-04-13 15:23:14: ERROR[Main]: stack traceback:
minetest_1  | 2020-04-13 15:23:14: ERROR[Main]:         /usr/local/share/minetest/builtin/game/falling.lua:106: in function 'set_node'
minetest_1  | 2020-04-13 15:23:14: ERROR[Main]:         /usr/local/share/minetest/builtin/game/falling.lua:178: in function </usr/local/sh
are/minetest/builtin/game/falling.lua:173>
```
##### Steps to reproduce

I really dont know kids were playing and then the server shut down.

### PR DESCRIPTION
**Pull request by SmallJoker**
Fixes #9659 by enforcing type consistency when they're provided (not nil).

## To do

This PR is Ready for Review.

## How to test

1) Launch any game you have installed
2) Create a temporary testing world
3) Enable any mob mod you can find
4) Run `/entities` and do the error tests below.
```Lua
minetest.register_chatcommand("entities", {
	func = function(pname)
		local pos = minetest.get_player_by_name(pname):get_pos()
		for name, def in pairs(minetest.registered_entities) do
			assert(minetest.add_entity(vector.add(pos, {
					x = math.random() * 6,
					y = math.random() * 6 + 8,
					z = math.random() * 6,
				}), name))
			print("Spawned " .. name)
		end
	end
})


if true then return end
-- Error tests

minetest.register_node(":errornode", {
	description = {},
	visual_scale = { x = 10, y = -1 },
	use_texture_alpha = "yes"
})

-- /spawnentity errorentity
minetest.register_entity(":errorentity", {
	initial_properties = {
		hp_max = function() end,
		is_visible = {},
		mesh = math.huge
	}
})
```

### CODE REVIEW

File: src/script/common/c_content.cpp [hunk 1]
```diff
@@ -1067,30 +1076,30 @@ NodeBox read_nodebox(lua_State *L, int index)
...
 
 #define NODEBOXREADVEC(n, s) \
-		lua_getfield(L, index, (s)); \
-		if (lua_istable(L, -1)) \
-			(n) = read_aabb3f_vector(L, -1, BS); \
-		lua_pop(L, 1);
+	lua_getfield(L, index, (s)); \
+	if (lua_istable(L, -1)) \
+		(n) = read_aabb3f_vector(L, -1, BS); \
+	lua_pop(L, 1);
```
Comment (sfan5) [inline src/script/common/c_content.cpp:1082]: Prime candidate for replacing with a lambda

File: src/script/common/c_converter.cpp [hunk 2]
```diff
@@ -457,12 +456,22 @@ size_t read_stringlist(lua_State *L, int index, std::vector<std::string> *result
 	Table field getters
 */
 
+bool check_field_or_nil(lua_State *L, int type, const char *fieldname)
```
Comment (sfan5) [inline src/script/common/c_converter.cpp:459]: Should have an `index` argument even if it's always `-1`

File: src/script/common/c_content.cpp (hunk 1 above)
Comment (SmallJoker) [reply src/script/common/c_content.cpp:1082]: Indeed, although it doesn't quite fit into the scope/goals of this PR. The lines were only re-indented.

Comment (sfan5) [reply src/script/common/c_content.cpp:1082]: Right.

File: src/script/common/c_converter.cpp (hunk 2 above)
Comment (SmallJoker) [reply src/script/common/c_converter.cpp]: added

**Pull request merged**